### PR TITLE
simple fixes

### PR DESCRIPTION
**Pull request by itsdiy0**
### Describe your change:
1. fixed a bug in the input of the function which was not  letting the code run
2. removed an input out of 2 inputs in the main function (the input was for the getting number of elements in the array and the second one was for the array itself, I removed the first one so now it just takes the array itself and calculates the length of array, more efficient and not hard coded)

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by realstealthninja — APPROVED** [review]

File: other/maximum_subarray.py [hunk 1]
```diff
@@ -21,6 +21,6 @@ def max_subarray(nums: list[int]) -> int:
 
 
 if __name__ == "__main__":
-    n = int(input("Enter number of elements : ").strip())
-    array = list(map(int, input("\nEnter the numbers : ").strip().split()))[:n]
-    print(max_subarray(array))
+    array = list(map(int, input("\nEnter the numbers : ").strip().split()))
+    n = len(array)
+    print(max_subarray(array[:n]))
```
Comment (CaedenPH) [inline other/maximum_subarray.py:26]: Whats the point of doing `array[:n]` or even creating variable `n`? All you're doing is saying "index this list up to all its elements"

File: other/maximum_subarray.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-def max_subarray(nums: list[int]) -> int:
+def max_subarray(nums: list) -> int:
```
Comment (CaedenPH) [inline other/maximum_subarray.py:1]: What is the point of this change?

Comment (itsdiy0) [reply other/maximum_subarray.py:1]: its getting type error with that and not running, so I just removed : 
Traceback (most recent call last):
  File "maxsubbarray.py", line 1, in <module>
    def max_subarray(nums: list[int]) -> int:
TypeError: 'type' object is not subscriptable

File: other/maximum_subarray.py (hunk 1 above)
Comment (itsdiy0) [reply other/maximum_subarray.py:26]: yea, You are right, n is not need at all

File: other/maximum_subarray.py (hunk 2 above)
Comment (CaedenPH) [reply other/maximum_subarray.py:1]: > its getting Typeerror with that and not running, so I just removed (try it) : Traceback (most recent call last): File "maxsubbarray.py", line 1, in def max_subarray(nums: list[int]) -> int: TypeError: 'type' object is not subscriptable

That is a problem with your python version. `list` being used for typehinting is only available on python versions after python `3.10`

Comment (itsdiy0) [reply other/maximum_subarray.py]: oh ok, I just removed the n in my commit then, I kept the type-hinting

**Pull request closed without merging**